src/test/utilities/promiseutilities.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import { PromiseSharer } from '../../shared/utilities/promiseUtilities'
+
+async function sleep(ms: number): Promise<void> {
+    return new Promise<void>(resolve => {
+        setTimeout(resolve, ms)
+    })
+}
+
+describe('PromiseSharer', async () => {
+    it('joins promises', async () => {
+        let timesCalled: number = 0
+        let promisesCompleted: number = 0
+
+        const doThing = async () => {
+            await sleep(150)
```

Comment:
Sleeps are unstable and slow down test execution. We should manually resolve the promises instead. I'm fine with keeping the sleeps for now, and adding a TODO to remove them.